client/signup/steps/difm-site-picker/index.tsx

```diff
@@ -0,0 +1,197 @@
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { getSiteDomain, getSiteSlug, getSiteTitle } from 'calypso/state/sites/selectors';
+import { SiteData } from 'calypso/state/ui/selectors/get-selected-site';
+
+interface Props {
+	stepSectionName: string | null;
+	stepName: string;
+	goToStep: () => void;
+	goToNextStep: () => void;
+}
+
+const DIFMSitePicker = ( {
+	filter,
+	onSiteSelect,
+}: {
+	filter: ( site: SiteData ) => boolean;
+	onSiteSelect: ( siteId: number ) => void;
+} ) => {
+	return (
+		<Card className="difm-site-picker__wrapper">
+			<SiteSelector filter={ filter } onSiteSelect={ onSiteSelect } />
+		</Card>
+	);
+};
+
+export default function DIFMSitePickerStep( props: Props ): React.ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const { goToNextStep } = props;
+	const [ siteId, setSiteId ] = useState< number | null >( null );
+	const [ confirmDomain, setConfirmDomain ] = useState( '' );
+	const siteDomain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const siteTitle = useSelector( ( state ) => getSiteTitle( state, siteId ) );
+	const headerText = translate( 'Choose where you want us to build your site.' );
+	const subHeaderText = translate( 'Some unsupported sites may be hidden.' );
+
+	useEffect( () => {
+		dispatch( saveSignupStep( { stepName: props.stepName } ) );
+	}, [ dispatch, props.stepName ] );
+
+	const handleSiteSelect = ( siteId: number ) => {
+		setSiteId( siteId );
+	};
+
+	const filterSites = ( site: SiteData ) => {
+		return (
+			site.capabilities?.manage_options &&
+			! site.jetpack &&
+			! site.options?.is_wpforteams_site &&
+			! site.options?.is_difm_lite_in_progress
+		);
+	};
+
+	const onCloseDialog = () => {
+		setSiteId( null );
+	};
+
+	const onConfirmDelete = () => {
+		dispatch(
+			submitSignupStep(
+				{
+					stepName: props.stepName,
+					stepSectionName: props.stepSectionName,
+					siteId,
+					siteSlug,
+				},
+				{
+					siteId,
+					siteSlug,
+				}
+			)
+		);
+		//Skip domains step
+		dispatch(
+			submitSignupStep(
+				{
+					stepName: 'domains',
+					wasSkipped: true,
+				},
+				{
+					domainItem: undefined,
+					themeItem: undefined,
+				}
+			)
+		);
+
+		goToNextStep();
+	};
+
+	if ( siteId ) {
+		const deleteDisabled = false;
+		// typeof confirmDomain !== 'string' ||
+		// confirmDomain.toLowerCase().replace( /\s/g, '' ) !== siteDomain;
+		const buttons = [
+			<Button onClick={ onCloseDialog }>{ translate( 'Cancel' ) }</Button>,
+			<Button primary scary disabled={ deleteDisabled } onClick={ onConfirmDelete }>
+				{ translate( 'Delete site content' ) }
+			</Button>,
+		];
+
+		return (
+			<Dialog isVisible={ true } buttons={ buttons } onClose={ onCloseDialog }>
+				<h1>{ translate( 'Site Reset Confirmation' ) }</h1>
+				<p>{ translate( 'After completing your purchase:' ) }</p>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **52** times:
`translate( 'Completing your purchase', { context: 'Loading state on /checkout'} )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-or-existing-step-in-difm)